api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -3014,4 +3011,32 @@ public void getWorkspaceResources() {
     assertThat(dataSets).hasSize(1);
     compareDatasetMetadata(dataSets.get(0), dataSet);
   }
+
+  @Test
+  public void testPublishCommunityWorkspace_ByUserWithWriterAccess() {
+    assertThrows(
+        ForbiddenException.class,
+        () -> {
+          Workspace ws = createWorkspace();
+          ws = workspacesController.createWorkspace(ws).getBody();
+          ws.setName("updated-name");
+          stubGetWorkspace(
+              ws.getNamespace(), ws.getId(), ws.getCreator(), WorkspaceAccessLevel.WRITER);
+          workspacesController.publishCommunityWorkspace(ws.getNamespace());
```

Comment:
nit: I prefer the style where `assertThrows()` only runs one command, so it's clearer which one throws.  Like this:
```
assertThrows(
        ForbiddenException.class,
        () -> workspacesController.publishCommunityWorkspace(ws.getNamespace()));
```